design/file-descriptors.md

```diff
@@ -0,0 +1,14 @@
+# File Descriptors
+File Descriptors ([witx type `fd`](https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#-fd)) are an abstract way of representing access to a resource. More often than not, they are processes (WIP), sockets (WIP), files, or even the console.
```

Comment:
This is a good start, and we can build from here. You're right that file descriptors are a foundational concept, and I expect over time we'll add on more text here. One thing that would be useful to say right now is that "file descriptors" are also sometimes called "handles", as we tend to use those terms somewhat interchangeably today.

Reply:
Since handles is a more accurate term here since these can refer to pretty much any kind of resource (especially when imported types are part of wasm), should the terminology be completely replaced by "handle"?

Reply:
Eventually, handles should be represented in wasm by references, at which point we can start being more careful to use the term "file descriptors" to just mean integer indices into the handle table.

Concerning the name "file descriptor" applying to things that aren't files, I agree it's confusing, however "file descriptor" and "fd" are commonly used in POSIX and Unix-type platforms for things that aren't files, eg. in [`socket`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/socket.html). "Everything is a file", after all ;-). Given that we'll hopefully be saying "handle" most of the time, especially at the witx / documentation level, I think it makes sense to continue using the POSIX "file descriptor" and "fd" names to the extent that we're talking about integer indices, though it's admittedly subjective.

Reply:
I made it more explicitly clear that they're interchangeable with `handle`s. I'm not familiar with the in detail workings of Wasm/Wasi to choose on my own if I should include the nuanced discussion about what an `fd`/`handle` is at its core, so I chose to leave that complexity out for the time being.